pytorch_lightning/trainer/training_loop.py

```diff
@@ -289,100 +283,75 @@ def training_step(self, split_batch, batch_idx, opt_idx, hiddens):
 
             # manually capture logged metrics
             model_ref._current_fx_name = 'training_step'
-            model_ref._results = Result()
             with self.trainer.profiler.profile("training_step"):
                 training_step_output = self.trainer.accelerator.training_step(step_kwargs)
                 self.trainer.accelerator.post_training_step()
 
-            self.trainer.logger_connector.cache_logged_metrics()
-
             training_step_output = self.trainer.call_hook("training_step_end", training_step_output)
 
             self._check_training_step_output(training_step_output)
 
-            training_step_output_for_epoch_end, training_step_output = self._process_training_step_output(
-                training_step_output, split_batch
-            )
-            if training_step_output_for_epoch_end is None:
+            training_step_output = self._process_training_step_output(training_step_output)
+            if training_step_output is None:
                 return
 
-        # enable empty loss when using manual opt
         closure_loss = None
-        untouched_loss = None
-
+        loss = None
         if self.trainer.lightning_module.automatic_optimization:
             # accumulate loss. if accumulate_grad_batches==1, no effect
             closure_loss = training_step_output.minimize / self.trainer.accumulate_grad_batches
-
             # the loss will get scaled for amp. avoid any modifications to it
-            untouched_loss = closure_loss.detach().clone()
-
-        # result
-        result = AttributeDict(
-            closure_loss=closure_loss,
-            loss=untouched_loss,
-            training_step_output=training_step_output,
-            training_step_output_for_epoch_end=training_step_output_for_epoch_end,
-        )
-        return result
-
-    def _process_training_step_output(self, training_step_output, split_batch):
-        training_step_output_for_epoch_end = training_step_output
+            loss = closure_loss.detach().clone()
+        return AttributeDict(closure_loss=closure_loss, loss=loss, training_step_output=training_step_output)
```

Comment:
Do we need the `AttributeDict` ?

Reply:
It's how the argument passing between loops was initially designed. I wouldn't want to modify this in this PR.

@awaelchli Have you kept them for the loops poc?

Should we replace them with a dataclass so the values are formalized?

Reply:
Yes we I think we should make a fixed return result. 
AttributeDict is currently used to also provide a signal if there was a reason to stop the loop early. 
I have not touched that part in the loop refactor.

Reply:
added a todo in https://github.com/PyTorchLightning/pytorch-lightning/pull/7700